lib/Extension/LanguageServerEvaluatableExpression/Protocol/EvaluatableExpression.php

```diff
@@ -0,0 +1,96 @@
+<?php // Auto-generated from vscode-languageserver-protocol (typescript)
+
+namespace Phpactor\Extension\LanguageServerEvaluatableExpression\Protocol;
+
+use DTL\Invoke\Invoke;
+use Exception;
+use RuntimeException;
+use Phpactor\LanguageServerProtocol\Range;
+
+class EvaluatableExpression
+{
+    public ?string $expression;
+
+    public Range $range;
+
+    public function __construct(Range $range, ?string $expression = null)
+    {
+        $this->expression = $expression;
+        $this->range = $range;
+    }
+
+    /**
+     * @param array<string,mixed> $array
+     * @return self
+     */
+    public static function fromArray(array $array, bool $allowUnknownKeys = false)
+    {
+        $map = [
+            'expression' => ['names' => [], 'iterable' => false],
+            'range' => ['names' => [Range::class], 'iterable' => false],
+        ];
+
+        foreach ($array as $key => &$value) {
+            if (!isset($map[$key])) {
+                if ($allowUnknownKeys) {
+                    unset($array[$key]);
+                    continue;
+                }
+
+                throw new RuntimeException(sprintf(
+                    'Parameter "%s" on class "%s" not known, known parameters: "%s"',
+                    $key,
+                    self::class,
+                    implode('", "', array_keys($map))
+                ));
+            }
+
+            // from here we only care about arrays that can be transformed into
+            // objects
+            if (!is_array($value)) {
+                continue;
+            }
+
+            if (empty($map[$key]['names'])) {
+                continue;
+            }
+
+            if ($map[$key]['iterable']) {
+                $value = array_map(function ($object) use ($map, $key, $allowUnknownKeys) {
```

Comment:
as this was generated code it can be simplified and the above condition _is_ always false. I would simplify this:

```
fromArray(array $array, bool $whatever) {
    return new self($array['one'], $array['two'], $array['three']);
}
```

and add some simple guard conditions to make phpstan happy.

Reply:
Yeah. Wanted to use the generator... That one is also out of date and I had to patch it first.. Will submit a PR there.

Perhaps you're right. I'll just clean up and hand code the minimum.

Reply:
I don't think `fromArray` is ever used, as this class is just used in LS->IDE direction... anyway...